community-events/content/challenge-project/02-exercise-swa-cli.md

```diff
@@ -0,0 +1,160 @@
+
+In this exercise, you'll learn how to run the **Portal** project in Codespaces and how to use the Azure Static Web Apps CLI to handle communication between the frontend and backend.
+
+## What is Azure Static Web App CLI?
+
+The **[Azure Static Web Apps (SWA) CLI](https://github.com/Azure/static-web-apps-cli/?WT.mc_id=academic-101248-cyzanon)** is an open-source command-line tool that simplifies local development and deployment to Azure Static Web Apps. It allows you to run your app locally or in a dev container, which in this case is GitHub Codespace, and then deploy your app to a production environment with just one command.
+
+## How does the communication between frontend and backend work?
+
+When we use the Azure Static Web CLI, we can establish communication between the Back-End and Front-End through the file: `swa-cli.config.json`.
+
+This file can be generated once you have globally installed the package: **[@azure/static-web-apps-cli](https://www.npmjs.com/package/@azure/static-web-apps-cli)**, by executing the following command inside the folder: **Portal**.
+
+```bash
+swa init
+```
+
+Please see the step-by-step creation of the file in the image below:
+
+![SWA init Contoso](./images/swa-init-contoso.gif)
+
+Once the file is generated according to the project settings, please observe in the code below that the `swa-cli.config.json` file has the following properties:
+
+<details><summary><b>packages/portal/swa-cli.config.json</b></summary>
+<br/>
+
+  ```json
+    {
+      "$schema": "https://aka.ms/azure/static-web-apps-cli/schema",
+      "configurations": {
+        "contoso-real-estate": {
+          "appLocation": ".",
+          "apiLocation": "../api",
+          "outputLocation": "dist/contoso-app",
+          "appBuildCommand": "npm run build",
+          "apiBuildCommand": "npm run build",
+          "run": "npm start",
+          "appDevserverUrl": "http://localhost:4200",
+          "apiDevserverUrl": "http://127.0.01:7071"
+        }
+      }
+    }
+  ```
+</details>
+<br/>
+
+Note the `appDevserverUrl` and `apiDevserverUrl` properties, which contain the paths to the frontend and backend, respectively. The `apiLocation` property contains the path to the `API` project. Onde se encontra o lado do Back-End da aplicação desenvolvido com: Azure Functions e Azure Cosmos DB (MongoDB API).
+
+## How to run the `portal` project?
+
+We do not recommend running this project locally, as it requires a number of configurations to work. To optimize your learning, we recommend using **[Codespaces](https://github.com/features/codespaces)** which will create a development container for you. And the best part of all: without you having to install anything on your machine.
+
+> Codespaces offers 60 free hours of usage per month. After this period, you will be charged an hourly rate. To learn more, visit [Codespaces](https://github.com/features/codespaces).
+
+To run the project, follow the steps below:
```

Comment:
steps 1-3 were already in the `01-prepare.md` file?

Reply:
you're right. I'm updating the steps!